tests/amsterdam/eip7928_block_level_access_lists/test_cases.md

```diff
@@ -2,17 +2,25 @@
-
-> ℹ️ Scope describes whether a test spans a single transaction (`tx`) or entire block (`blk`).
+| `test_bal_7702_delegated_create` | Ensure BAL tracks EIP-7702 delegation indicator write and contract creation | Alice sends a type-4 (7702) tx authorizing herself to delegate to `Deployer` code which executes `CREATE` | BAL **MUST** include for **Alice**: `code_changes` (delegation indicator), `nonce_changes` (increment from 7702 processing), and `balance_changes` (post-gas). For **Child**: `code_changes` (runtime bytecode) and `nonce_changes = 1` | 🟡 Planned |
+| `test_bal_self_transfer` | Ensure BAL handles self-transfers correctly | Alice sends `1 ETH` to **Alice** | BAL **MUST** include **one** entry for Alice with `balance_changes` reflecting **gas only** (value cancels out) and a nonce change; Coinbase balance updated for fees; no separate recipient row | 🟡 Planned |
+| `test_bal_system_contracts_2935_4788` | Ensure BAL includes pre-exec system writes for parent hash & beacon root | Build a block with `N` normal txs; 2935 & 4788 active | BAL **MUST** include `HISTORY_STORAGE_ADDRESS` (EIP-2935) and `BEACON_ROOTS_ADDRESS` (EIP-4788) with `storage_changes` to ring-buffer slots; each write uses `tx_index = N` (system op) | 🟡 Planned |
+| `test_bal_system_dequeue_withdrawals_eip7002` | Ensure BAL tracks post-exec system dequeues for withdrawals | Pre-populate EIP-7002 withdrawal requests; produce a block where dequeues occur | BAL **MUST** include the 7002 system contract with `storage_changes` (queue head/tail slots 0–3) using `tx_index = len(txs)` and balance changes for withdrawal recipients | 🟡 Planned |
+| `test_bal_system_dequeue_consolidations_eip7251` | Ensure BAL tracks post-exec system dequeues for consolidations | Pre-populate EIP-7251 consolidation requests; produce a block where dequeues occur | BAL **MUST** include the 7251 system contract with `storage_changes` (queue slots 0–3) using `tx_index = len(txs)` | 🟡 Planned |
+| `test_bal_noop_storage_write` | Ensure BAL includes address with storage read for no-op writes where pre-state equals post-state | Contract with pre-existing storage value `0x42` in slot `0x01`; transaction executes `SSTORE(0x01, 0x42)` | BAL **MUST** include the contract address with `storage_reads` for the slot since it was accessed, but **MUST NOT** include it in `storage_changes` | 🟡 Planned |
+| `test_bal_oog_intrinsic` | Ensure BAL handles intrinsic out-of-gas (OOG) failure correctly | Transaction with gas limit below intrinsic cost | BAL **MUST** be empty for transactions that fail intrinsic gas validation | 🟡 Planned |
```

Comment:
Also as I mentioned in the other PR, perhaps insufficient-intrinsic-gas is not the correct approach to test since,  normally, exception tests obscure other checks, so any block-level access list check will be irrelevant when another verification failed earlier during the block verification process (i.e. we won't even get to the part where the BAL is checked because the block contained an invalid transaction).

I think the approach should be to test OOG, reverts, invalid-opcode, or other types of EVM execution aborts, and verify that the BAL, when something like that happens during the EVM execution of a transaction that was successfully included in the block, is still correctly calculated.